.github/workflows/unit-test.yml

```
@@ -97,61 +102,132 @@ jobs:
         with:
           python-version: '${{ inputs.python_version }}'
 
       - name: Upgrade PIP
+        if: steps.continue.outputs.confirm == 'true'
         run: pip install --upgrade pip && pip install wheel --upgrade
 
       - name: Install PennyLane dependencies
+        if: steps.continue.outputs.confirm == 'true'
         run: |
           pip install -r requirements-ci.txt --upgrade
           pip install -r requirements-dev.txt --upgrade
 
       - name: Install PyTorch
-        if: inputs.install_pytorch == true
+        if: inputs.install_pytorch == true && steps.continue.outputs.confirm == 'true'
         env:
           TORCH_VERSION: ${{ inputs.pytorch_version != '' && format('=={0}', inputs.pytorch_version) || '' }}
         run: pip install "torch${{ env.TORCH_VERSION }}" -f https://download.pytorch.org/whl/torch_stable.html
 
       - name: Install TensorFlow
-        if: inputs.install_tensorflow == true
+        if: inputs.install_tensorflow == true && steps.continue.outputs.confirm == 'true'
         env:
           TF_VERSION: ${{ inputs.tensorflow_version != '' && format('~={0}', inputs.tensorflow_version) || '' }}
         run: pip install "tensorflow${{ env.TF_VERSION }}" "keras${{ env.TF_VERSION }}"
 
       - name: Install JAX
-        if: inputs.install_jax == true
+        if: inputs.install_jax == true && steps.continue.outputs.confirm == 'true'
         env:
           JAX_VERSION: ${{ inputs.jax_version != '' && format('=={0}', inputs.jax_version) || '' }}
         run: pip install "jax${{ env.JAX_VERSION}}" "jaxlib${{ env.JAX_VERSION }}"
 
       - name: Install additional PIP packages
-        if: inputs.additional_pip_packages != ''
+        if: inputs.additional_pip_packages != '' && steps.continue.outputs.confirm == 'true'
         run: pip install ${{ inputs.additional_pip_packages }}
 
       - name: Install PennyLane
+        if: steps.continue.outputs.confirm == 'true'
         run: |
           python setup.py bdist_wheel
           pip install dist/PennyLane*.whl
 
       - name: Install PennyLane-Lightning master
-        if: inputs.install_pennylane_lightning_master == true
+        if: inputs.install_pennylane_lightning_master == true && steps.continue.outputs.confirm == 'true'
         run: pip install -i https://test.pypi.org/simple/ PennyLane-Lightning --pre --upgrade
 
+      - name: Set Benchmark Args
+        id: benchmark_args
+        if: contains(inputs.pipeline_mode, 'benchmarks')
+        run: echo "args=--benchmark-enable --benchmark-only --benchmark-json=benchmarks.json" >> $GITHUB_OUTPUT
+
+      - name: Set Coverage Args
+        id: coverage_args
+        if: inputs.pipeline_mode == 'unit-tests'
+        run: echo "args=${{ inputs.pytest_coverage_flags }}" >> $GITHUB_OUTPUT
+
+      - name: Set Parallelization Args
+        id: parallel_args
+        if: inputs.pipeline_mode == 'unit-tests'
+        run: echo "args=-n auto" >> $GITHUB_OUTPUT
+
```

Though nothing inherently wrong with this implementation, I was thinking since there are three steps now, it might be easier to have one step that sets pytest args in general:
```suggestion
      - name: Set PyTest Args
        id: pytest_args
        env:
          PIPELINE_MODE: ${{ inputs.pipeline_mode }}
          PYTEST_COVERAGE_ARGS: ${{ inputs.pytest_coverage_flags }}
	  PYTEST_PARALLELISE_ARGS: -n auto
	  PYTEST_BENCHMARKS_ARGS: --benchmark-enable --benchmark-only --benchmark-json=benchmarks.json
	  PYTEST_ADDITIONAL_ARGS: ${{ inputs.pytest_additional_args }}
        run: |
          if [[ "$PIPELINE_MODE" =~ .*"benchmarks".* ]]; then
	    echo "args=$PYTEST_COVERAGE_ARGS $PYTEST_PARALLELISE_ARGS $PYTEST_ADDITIONAL_ARGS" >> $GITHUB_OUTPUT
	  else
	    echo "args=$PYTEST_BENCHMARKS_ARGS $PYTEST_ADDITIONAL_ARGS" >> $GITHUB_OUTPUT
	  fi
```